module/modmanager/manager.go

```diff
@@ -996,11 +956,25 @@ func (m *module) dial() error {
 	if err != nil {
 		return errors.WithMessage(err, "module startup failed")
 	}
-	m.conn.ReplaceConn(rpc.GrpcOverHTTPClientConn{ClientConn: conn})
-	m.client = pb.NewModuleServiceClient(&m.conn)
+
+	// Take the grpc over unix socket connection and add it to this `module`s `SharedConn`
+	// object. This `m.sharedConn` object is referenced by all resources/components. `Client`
+	// objects communicating with the module. If we're re-dialing after a restart, there may be
+	// existing resource `Client`s objects. Rather than recreating clients with new information, we
+	// choose to "swap out" the underlying connection object for those existing `Client`s.
+	//
+	// Resetting the `SharedConn` will also create a new WebRTC PeerConnection object. `dial`ing to
+	// a module is followed by doing a `ReadyRequest` `ReadyResponse` exchange. If that exchange
+	// contains a working WebRTC offer and answer, the PeerConnection will succeed in connecting. If
+	// there is an error exchanging offers and answers, the PeerConnection object will be nil'ed
+	// out.
+	m.sharedConn.ResetConn(rpc.GrpcOverHTTPClientConn{ClientConn: conn}, m.logger)
```

Comment:
I don't think we ever initialize `m.sharedConn` do we? Won't it always be a nil ptr & therefor always panic?

Reply:
never mind